### PR TITLE
add docs for menu's .Title and .Page variables

### PR DESCRIPTION
**Pull request by rmetzler**
This is documentation for gohugoio/hugo#4160

I split this into two commits. The first one is the actual documentation, the second one is an example usage in some of the pages. The second commit needs a hugo version >= `v0.32`.

I had difficulties to create and test a TOML example. Therefore I only include YAML examples.

Here is what I tried

```md
+++
title="TOML Example"
date=2017-12-16
linktitle="TOML Example"
[[menu.docs]]
  title="how to use menus in templates"
  parent="templates"
+++

I don't understand TOML.
```

### CODE REVIEW

**Comment by kaushalmodi** [comment]
Thank you for this!

As the documentation is for a new feature in a Hugo version that's not yet released, the PR should go in the hugo repo itself, not the hugoDocs repo. The hugoDocs repo is for latest hugo release, because as soon as the commit here gets merged, it goes live on the hugoDocs site. 

By making this a PR in the hugo repo, you don't need to add disclaimers like "this will work only in Hugo >=v0.32". 

@bep merges the doc changes in the hugo repo in to this repo at the time of each release. So your doc changes with eventually make it on the live site then.

**Comment by bep** [comment]
@kaushalmodi let us just let this PR live here and we can postpone the merge.

**Comment by kaushalmodi** [comment]
Ok

**Comment by rmetzler** [comment]
Ok, obvoiusly I didn't understand your comment regarding the repo as intended.

Just let me know if you need anything else from me. I can also move this PR over to the gohugoio/hugo repo.

File: content/templates/menu-templates.md [hunk 1]
```diff
@@ -103,3 +108,55 @@ The above is all that's needed. But if you want custom menu items, e.g. changing
...
+...
+```
+
+{{% note %}}
+You can define more than one menu. It also doesn't have to be a complex value,
+`menu` can also be a string, an array of strings, or an array of complex values
+like in the example above.
+{{% /note %}}
+
+### Using .Page in Menues
```
Comment (kaushalmodi) [inline content/templates/menu-templates.md]: Typo: Menues -> Menus

File: content/templates/menu-templates.md [hunk 2]
```diff
@@ -103,3 +108,55 @@ The above is all that's needed. But if you want custom menu items, e.g. changing
...
+
+If you use the front matter method of defining menu entries, you'll get access to the .Page variable.
```
Comment (kaushalmodi) [inline content/templates/menu-templates.md]: .Page -> `` `.Page` `` for consistency

**Comment by kaushalmodi** [comment]
> Just let me know if you need anything else from me. I can also move this PR over to the gohugoio/hugo repo.

There's nothing else that needs to be done.. Just that this PR will get merged after the next hugo release (Note: @digitalcraftsman ).

**Comment by rmetzler** [comment]
Hi, I just wanted to remind you for the merge. Thanks for releasing Hugo 0.32!

**Pull request merged**